blamo/lib/PDF/pdf_builder.dart

```diff
@@ -128,94 +122,122 @@ List<Level> trimSplitAtIndex(List<Level> splits, int i, double maxHeight){
 }
 
 
-List<Level> boxSplit(Level l, double maxHeight){ 
+List<Level> boxSplit(Level l, double maxHeight){        // manually divide a level whose rendered height exceeds maxHeight over multiple new valid levels. returns a list of these new levels
   List<Level> splits = [];
-  splits.add(l);
-  for (int i = 0; i < splits.length; i++){
-    if (splits[i].tests.length > 0){
-      splits = trimSplitAtIndex(splits, i, maxHeight);
+  splits.add(l);                                        // init our new smaller level array with our base overflowing level.
+  for (int i = 0; i < splits.length; i++){              // while we have splits left,
+    if (splits[i].tests.length > 0){                    //  if there are tests in the level,
+      splits = trimSplitAtIndex(splits, i, maxHeight);  //    trim all overflowing tests into a new level ("split")
     }
-    else{
-      splits.remove(splits[i]);
+    else{                                               //  if there aren't any tests in the level,
+      splits.remove(splits[i]);                         //    remove it for the sake of condensing whitespace.
     }
   }
 
   return splits;
 }
 
+void addFailed(String e){                               // Override the pdf with an error pdf. Body text passed as argument.
```

Comment:
If this is something we can avoid i might say to remove the adding of a failed pdf, and just return an error to the user. Having an ERROR pdf might cause issues if a user expects to email something that isn't working or is only partially correct.